presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcess.java

```diff
@@ -148,6 +149,18 @@ public void close()
     {
         if (process != null && process.isAlive()) {
             process.destroy();
+            try {
+                process.waitFor(120, TimeUnit.SECONDS);
```

Comment:
Wondering how we arrived at this number 120 seconds.. ? Is it just by experimentation ?

Reply:
120 seconds is because: SHUTDOWN itself takes 10 seconds wait time. Periodic task cleanning is every 60 seconds. So doulbling the time before force killing. I can add some comments